test/integration/Cover/expireCover.js

```diff
@@ -0,0 +1,612 @@
+
+    // enable active cover tracking
+    await cover.connect(emergencyAdmin).enableActiveCoverAmountTracking([], []);
+    await cover.connect(emergencyAdmin).commitActiveCoverAmounts();
+
+    // Buy Cover
+    await buyCover({
+      ...expireCoverFixture,
+      targetPrice: DEFAULT_PRODUCT_INITIALIZATION[0].targetPrice,
+      cover,
+      coverBuyer1,
+    });
+
+    // submit incident
+    await submitIncident({ yc, productId, period, signer: this.accounts.defaultSender });
+
+    // accept incident
+    await as.connect(staker1).castVotes([0], [true], parseEther('100'));
+
+    {
+      // advance past payout cooldown
+      const { payoutCooldownInDays } = await as.config();
+
+      const { end } = await as.getPoll(0);
+      await setTime(end + daysToSeconds(payoutCooldownInDays));
+    }
+
+    // partial paid cover
+    const claimAmount = amount.div(2);
+    await yc.connect(coverBuyer1).redeemPayout(0, 0, 0, claimAmount, nonMember1.address, []);
+
+    {
+      // advance past expire time
+      const { timestamp: currentTime } = await ethers.provider.getBlock('latest');
+      await setTime(currentTime + period + 1);
+    }
+
+    {
+      const segment = await cover.coverSegments(coverId, segmentId);
+      expect(segment.expired).to.be.equal(false);
+
+      const activeCoverAmount = await cover.totalActiveCoverInAsset(coverAsset);
+      expect(activeCoverAmount).to.be.equal(segment.amount);
+    }
+
+    await cover.expireCover(coverId);
+
+    {
+      const segment = await cover.coverSegments(coverId, segmentId);
+      expect(segment.expired).to.be.equal(true);
+
+      const activeCoverAmount = await cover.totalActiveCoverInAsset(coverAsset);
+      expect(activeCoverAmount).to.be.equal(0);
+    }
+  });
+
+  it('expire a cover that had a partial claim paid out, cover tracking enabled after cover buy', async function () {
+    const { DEFAULT_PRODUCT_INITIALIZATION } = this;
+    const { cover, stakingPool0, as, yc, ybETH } = this.contracts;
+    const [coverBuyer1, staker1] = this.accounts.members;
+    const [nonMember1] = this.accounts.nonMembers;
+    const { emergencyAdmin } = this.accounts;
+
+    const { productId, coverAsset, period, gracePeriod, amount, coverId, segmentId } = expireCoverFixture;
+
+    // Stake to open up capacity
+    await stake({ stakingPool: stakingPool0, staker: staker1, productId, period, gracePeriod });
+
+    // cover buyer gets yield token
+    await transferYieldToken({ tokenOwner: this.accounts.defaultSender, coverBuyer1, yc, ybETH });
+
+    // Buy Cover
+    await buyCover({
+      ...expireCoverFixture,
+      targetPrice: DEFAULT_PRODUCT_INITIALIZATION[0].targetPrice,
+      cover,
+      coverBuyer1,
+    });
+
+    {
+      const segment = await cover.coverSegments(coverId, segmentId);
+
+      // NOTE: the active cover amount for the cover asset has to be correctly set, otherwise it will cause
```

Comment:
yes that's a fair assumption that the *final* values commited for active cover amount per asset are correct and the responsibility of the team.